metis-indexing/src/main/java/eu/europeana/indexing/IndexerImpl.java

```diff
@@ -134,29 +140,69 @@ public boolean indexTombstone(String rdfAbout) throws IndexerRelatedIndexingExce
     final FullBeanImpl publishedFullbean = this.connectionProvider.getIndexedRecordAccess().getFullbean(rdfAbout);
     if (publishedFullbean != null) {
       final FullBeanPublisher publisher = connectionProvider.getFullBeanPublisher(true);
-      prepareTombstoneFullbean();
+      final FullBeanImpl tombstoneFullbean = prepareTombstoneFullbean(publishedFullbean);
       try {
-        publisher.publishTombstone(publishedFullbean, publishedFullbean.getTimestampCreated());
+        publisher.publishTombstone(tombstoneFullbean, tombstoneFullbean.getTimestampCreated());
       } catch (IndexingException e) {
         throw new IndexerRelatedIndexingException("Could not create tombstone record '" + rdfAbout + "'.", e);
       }
     }
     return publishedFullbean != null;
   }
 
-  //TODO: 2024-08-29 - Once tombstones are working, we need to refine the fields we need. For now unused.
-  private FullBeanImpl prepareTombstoneFullbean() {
+  private FullBeanImpl prepareTombstoneFullbean(FullBeanImpl publishedFullbean) {
     final FullBeanImpl tombstoneFullbean = new FullBeanImpl();
-    final ChangeLog changeLog = new ChangeLogImpl();
-    changeLog.setType("Delete");
-    changeLog.setContext("http://data.europeana.eu/vocabulary/depublicationReason/noPermission");
-    changeLog.setEndTime(new Date());
-    final EuropeanaAggregation europeanaAggregation = new EuropeanaAggregationImpl();
-    europeanaAggregation.setChangeLog(List.of(changeLog));
-    tombstoneFullbean.setEuropeanaAggregation(europeanaAggregation);
+    tombstoneFullbean.setAbout(publishedFullbean.getAbout());
+    tombstoneFullbean.setTimestampCreated(publishedFullbean.getTimestampCreated());
+    tombstoneFullbean.setTimestampUpdated(publishedFullbean.getTimestampUpdated());
+
+    tombstoneFullbean.setEuropeanaAggregation(prepareEuropeanaAggregation(publishedFullbean.getEuropeanaAggregation()));
+    tombstoneFullbean.setAggregations(List.of(prepareAggregation(publishedFullbean.getAggregations().getFirst())));
+    final Optional<ProxyImpl> providerProxy =
+        publishedFullbean.getProxies().stream().filter(not(ProxyImpl::isEuropeanaProxy)).findFirst();
+    providerProxy.ifPresent(proxy -> tombstoneFullbean.setProxies(List.of(prepareProxy(proxy))));
     return tombstoneFullbean;
   }
 
+  private static EuropeanaAggregation prepareEuropeanaAggregation(EuropeanaAggregation europeanaAggregation) {
+    final ChangeLog tombstoneChangeLog = new ChangeLogImpl();
+    tombstoneChangeLog.setType("Delete");
+    tombstoneChangeLog.setContext("http://data.europeana.eu/vocabulary/depublicationReason/noPermission");
```

Comment:
this has to be parametrizable? the de-publication reason I mean.

Reply:
Yes it should be. As I mentioned in the standup I realized that after looking at the upcoming ticket MET-6060, so I will handle that with that ticket.